reference-implementation/lib/abstract-ops/readable-streams.js

```diff
@@ -1341,13 +1360,22 @@ function ReadableByteStreamControllerFillPullIntoDescriptorFromQueue(controller,
   const maxBytesToCopy = Math.min(controller._queueTotalSize,
                                   pullIntoDescriptor.byteLength - pullIntoDescriptor.bytesFilled);
   const maxBytesFilled = pullIntoDescriptor.bytesFilled + maxBytesToCopy;
-  const maxAlignedBytes = maxBytesFilled - maxBytesFilled % elementSize;
 
   let totalBytesToCopyRemaining = maxBytesToCopy;
   let ready = false;
-  if (maxAlignedBytes > currentAlignedBytes) {
-    totalBytesToCopyRemaining = maxAlignedBytes - pullIntoDescriptor.bytesFilled;
-    ready = true;
+  if (pullIntoDescriptor.readFully) {
+    // A descriptor for a readFully() request that is not yet completely filled will stay at the head of the queue,
+    // so the underlying source can keep filling it.
+    if (maxBytesFilled >= pullIntoDescriptor.byteLength) {
+      totalBytesToCopyRemaining = pullIntoDescriptor.byteLength - pullIntoDescriptor.bytesFilled;
+      ready = true;
+    }
+  } else {
+    const maxAlignedBytes = maxBytesFilled - maxBytesFilled % elementSize;
+    if (maxAlignedBytes > currentAlignedBytes) {
```

Comment:
Can't this be simplified to `maxAlignedBytes > 0`? 🤔 If the pull-into descriptor was still in the queue, then it must have `bytesFilled < elementSize`. So `currentAlignedBytes = bytesFilled - bytesFilled % elementSize` must also be `0`. (I feel like I never understood why `currentAlignedBytes` was necessary.)

All tests pass fine when I add an `assert(currentAlignedBytes === 0)` here, so I think this is valid? Unless I'm missing an edge case (that isn't covered by any WPT test)?

Reply:
I'm pretty sure this is right, so I changed it. I added some asserts for good measure. 😁